File: String/PalindromeCheck.cpp
#include <iostream>
#include <string>
using namespace std;

int main(){
    string inputString;
    cin >> inputString;
    
    int left = 0;
    int right = inputString.size() - 1;
    
    bool isPalindrome = true;
    while(left < right){
        if(inputString.at(left) != inputString.at(right)) {
            isPalindrome = false;
            break;
        }
    }
    
    if(isPalindrome) cout << "String is a Palindrome.";
    else cout << "String is not a palindrome";
    
    return 0;
}

File: String/Java/PalindromeCheck.java
java.util.*;

class PalindromeCheck {
    public static void main(String args[]){
        Scanner input = new Scanner(System.in);
        String inputString = input.nextLine();
        char[] chars = inputString.toLowerCase().toCharArray();
        
        int left = 0;
        int right = chars.length - 1;
        
        boolean isPalindrome = true;
        
        while(left < right){
            if(chars[left] != chars[right]) {
                isPalindrome = false;
                break;
            }
            left++;
            right--;
        }
        
        if(ispalindrome) System.out.println("A Palindrome.");
        else System.out.println("Not a Palindrome.");
        
        input.close();
    }
}